lib/quadrigacx/client/private.rb

```diff
@@ -56,44 +56,26 @@ def open_orders params={}
       request(:post, '/open_orders', params)
     end
 
-    # Withdraw bitcoins.
+    # Withdrawal of the specified coin type.
     #
+    # coin    – The coin type
     # amount  – The amount to withdraw.
-    # address – The bitcoin address we will send the amount to.
-    def bitcoin_withdraw params={}
-      request(:post, '/bitcoin_withdrawal', params)
+    # address – The coin type's address we will send the amount to.
+    def withdraw coin, params={}
+      raise ConfigurationError.new('No coin type specified') unless coin
+      raise ConfigurationError.new('Invalid coin type specified') unless Coin.valid?(coin)
```

Comment:
Have you considered combining these two checks into one inside the `valid?` call? I.e. doing `unless coin` inside `.valid?`?

Reply:
I do like having the separate guards as its possible to raise the specific error given the condition.

If that is what you meant by combining them?

Reply:
Ok thats fair. Works for me.